src/internal/docker/cli.go

```diff
@@ -1,38 +1,100 @@
 package docker
 
 import (
+	"context"
 	"fmt"
+	"os"
+	"os/exec"
+	"strings"
 
+	"github.com/mitchellh/go-homedir"
 	yey "github.com/silphid/yey/src/internal"
 )
 
 type CLI struct{}
 
-func (c CLI) Start(ct yey.Context, imageTag, containerName string) error {
-	return fmt.Errorf("not implemented")
+func NewCLI() CLI {
+	return CLI{}
+}
 
-	// Get running ID and state
-	// docker ps --all --filter "name=al" --format '{{.ID}}|{{.State}}'
-	// state="exited"|"running"
+func (c CLI) Start(ctx context.Context, yeyCtx yey.Context, containerName string) error {
+	// Determine whether we need to run or exec container
+	isRunning, err := isContainerRunning(containerName)
+	if err != nil {
+		return err
+	}
+	args := []string{"run"}
+	if !isRunning {
+		args = append([]string{"exec"}, args...)
+	}
 
-	// Run
-	// docker run -it --name al alpine
+	// Determine extra docker arguments
+	extraArgs, err := getDockerArgs(yeyCtx, containerName, !isRunning)
+	if err != nil {
+		return err
+	}
+	args = append(args, extraArgs...)
 
-	// Exec
-	// run docker exec -it "${DOCKER_EXEC_ARGS[@]}" "${DOCKER_CONTAINER}" zsh
+	// Run docker command
+	args = append(args, yeyCtx.Cmd...)
+	cmd := exec.Command("docker", args...)
+	return cmd.Run()
 }
 
-// type containerState string
+func (cli CLI) ListContainers(ctx context.Context) ([]string, error) {
+	cmd := exec.Command("docker", "ps", "--all", "--filter", "name=yey-*", "--format", "{{.Name}}")
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	return getOutputLines(output), nil
+}
 
-// const (
-// 	stateRunning containerState = "running"
-// 	stateExited                 = "exited"
-// )
+func isContainerRunning(name string) (bool, error) {
+	cmd := exec.Command("docker", "ps", "--filter", "name="+name, "--format", "foo")
+	output, err := cmd.Output()
+	if err != nil {
+		return false, err
+	}
+	lines := getOutputLines(output)
+	return len(lines) > 0, nil
+}
+
+func getOutputLines(value []byte) []string {
+	return strings.Split(strings.ReplaceAll(string(value), "\r\n", "\n"), "\n")
+}
 
-// type container struct {
-// 	state containerState
-// }
+func getDockerArgs(yeyCtx yey.Context, containerName string, isExec bool) ([]string, error) {
+	// Common args
+	args := []string{"-it", "--env LS_COLORS", "--env TERM", "--env TERM_COLOR", "--env TERM_PROGRAM"}
+	if !isExec {
+		args = append(args, "--rm", "--name", containerName)
```

Comment:
TODO Discuss the philosophy of --rm

Reply:
Y'a tellement de flags docker possibles et j'aimerais mieux pas tous les hardcoder si possible (genre "--network=host" qui est très pratique, mais pas nécessairement ce que tout le monde veut). Alors, je pensais pouvoir configurer des flags explicitement dans le contexte. Comme ça, tout le monde pourrait contrôler ces flags qui sont plus personnels.